### DISCUSSION TITLE
unable to edit number of components in SIMPLISMA function

### DISCUSSION THREAD

**Discussion by GhostDeini** [spectrochempy/spectrochempy#502] [General]
### What happened?

Hi, I am trying to do a SIMPLISMA analysis but I am not able to obtain more than 2 components, even if I explicitly set n_pc, like this:

pure = scp.SIMPLISMA(array,n_pc=3,noise=10,tol=0.01)

I am able to do it in the interactive mode but I do not really get how it works. Could you please fix this bug? Thank you!

**Comment by atravert** [comment]
Hi, 

Can you attach the dataset on which you would like to carry out SIMPLISMA ?, I will look at thus issue.

**Comment by GhostDeini** [comment]
Hi, this is the array. It is composed of 229 XAS spectra. [cell4pos1.txt](https://github.com/spectrochempy/spectrochempy/files/10175964/cell4pos1.txt)
This is the energy scale (tho I don't think you need it for now)
[energy.txt](https://github.com/spectrochempy/spectrochempy/files/10175968/energy.txt)

Thank you again.

**Comment by atravert** [comment]
Hello,

The reason is that most of the variance of your dataset is explained by two components, so that goodness of fit criteria are reached with 2 components:   

```python
>>> sim_1 = scp.SIMPLISMA(D, n_pc=3)
>>> print(sim_1.logs)

*** Automatic SIMPL(I)SMA analysis *** 
dataset: NDDataset_ea0df440
  noise:  3 %
    tol: 0.1 %
   n_pc:  3
#iter index_pc  coord_pc   Std(res)   R^2   
---------------------------------------------
   1    262    8342.8     0.1597     0.9687 
   2    328    8356.0     0.0052     1.0000 
**** Unexplained variance lower than 'tol' (0.1%) 
**** End of SIMPL(I)SMA analysis.
``` 

To get more components, you could decrease the 'noise' parameter to 0.1% for instance:  

```python
>>> sim_2 = scp.SIMPLISMA(D, n_pc=3, noise=0.1)
>>> print(sim_2.logs)

*** Automatic SIMPL(I)SMA analysis *** 
dataset: NDDataset_6d95a662
  noise: 0.1 %
    tol: 0.1 %
   n_pc:  3
#iter index_pc  coord_pc   Std(res)   R^2   
---------------------------------------------
   1     31    8215.0     0.1324     0.9787 
   2     18    8150.0     0.1190     0.9829 
   3     25    8185.0     0.1028     0.9873 
**** Reached maximum number of pure compounds 'n_pc' (3) 
**** End of SIMPL(I)SMA analysis.
```


As it is not a bug, I move this issue to the discussion section, but let me know if you have further question about this !

Best regards
